### DISCUSSION TITLE
Thoughts on CSS-in-JS

### DISCUSSION THREAD

**Discussion by debel27** [mantinedev/mantine#3683] [Feedback]
There's been some discussion recently regarding the place of CSS-in-JS in the React ecosystem. Here are some relevant links on the topic

  - https://dev.to/srmagura/why-were-breaking-up-wiht-css-in-js-4g9b
  - https://twitter.com/dan_abramov/status/1584835028986793984

I was wondering where the Mantine team stands on this. Are alternatives to CSS-in-JS being considered ?

I realize migrating away from Emotion is a big undertaking and the decision should not be taken lightly. From what I understand, the Mantine team is [not planning on doing it](https://github.com/mantinedev/mantine/discussions/3579#discussioncomment-5047179) unless React gives them a clear incentive to do so (i.e., removes support). All I can ask is to reconsider.

With v18, React is entering a new age. The focus of the next years will be on frameworks and more opinionated practices regarding application development (such as Suspense-based architectures, Server components, etc.). I don't see how CSS-in-JS fits in that picture, where the emphasis will be on UX. It just seems like the wrong horse to bet on, even with mitigation like `useInsertionEffect` and static extraction.

I'd love to hear your thoughts on this.

**Comment by rtivital** [comment]
I think my comment says it clearly: if it is required for frameworks support, Mantine will use native CSS instead of css-in-js. I'm not really sure what else I can say about it.
